source/common/router/router.cc

```diff
@@ -392,8 +395,11 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
 
   ENVOY_STREAM_LOG(debug, "router decoding headers:\n{}", *callbacks_, headers);
 
-  UpstreamRequestPtr upstream_request = std::make_unique<UpstreamRequest>(*this, *conn_pool);
-  upstream_request->moveIntoList(std::move(upstream_request), upstream_requests_);
+  if (upstream_requests_.front() == nullptr) {
```

Comment:
This change is confusing to me. Don't we need to make an upstream request when we get headers? I'm trying to reconcile with `decodeMetadata` which seems to indicate that metadata must come after headers.

Reply:
Oops! Sorry, this change was made when I tried to allow metadata to create upstream request when they arrive before headers. But it turned out metadata are always behind headers, and I forgot to revert this change.